### PR TITLE
Ten 121

### PR DESCRIPTION
**Pull request by Chrissystk**
app-config-base https://github.com/hmrc/app-config-base/pull/7046
journey tests https://github.com/hmrc/tax-enrolment-assignment-journey-tests/pull/28
https://github.com/hmrc/app-config-qa/pull/14873
https://github.com/hmrc/app-config-staging/pull/10350
https://github.com/hmrc/app-config-production/pull/14443

### CODE REVIEW

File: it/controllers/AccountCheckControllerISpec.scala [hunk 1]
```diff
@@ -140,7 +140,7 @@ class AccountCheckControllerISpec extends IntegrationSpecBase with Status {
       }
 
       "has SA enrolment on an other account" should {
-        s"return OK" in {
+        s"retdirect to blue interupt page" in {
```
Comment (ChrisHillCap) [inline it/controllers/AccountCheckControllerISpec.scala:143]: controller reverse route

File: it/controllers/SABlueInterruptControllerISpec.scala [hunk 2]
```diff
@@ -18,33 +18,404 @@ package controllers
...
+          .get()
+
+        whenReady(res) { resp =>
+          resp.status shouldBe OK
+          resp.body should include("There was a problem")
+        }
+      }
+    }
+
+    "an authorised user with no credential uses the service" should {
```
Comment (ChrisHillCap) [inline it/controllers/SABlueInterruptControllerISpec.scala:125]: am I right in thinking alot of these tests below are generic, I added a comment in the other ISPEC you added, perhaps we should split these out and make a helper? if its possible

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 3]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
           .text() shouldBe "selfAssessmentInterrupt.paragraph1 " ++ "selfAssessmentInterrupt.paragraph2"
       }
     }
+
+    "the user does not have an account type of SA_ASSIGNED_TO_OTHER_USER" should {
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:91]: enum

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 4]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+          .expects(List(SA_ASSIGNED_TO_OTHER_USER), *, *, *)
+          .returning(createInboundResultError(InvalidUserType(Some("/test"))))
+
+        val result = controller
+          .view()
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
+
+        status(result) shouldBe SEE_OTHER
+        redirectLocation(result) shouldBe Some(
+          "/protect-tax-info?redirectUrl=%2Ftest"
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:120]: controller reverse route

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 5]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+            _: RequestWithUserDetails[AnyContent],
+            _: HeaderCarrier,
+            _: ExecutionContext
+          ))
+          .expects(List(SA_ASSIGNED_TO_OTHER_USER), *, *, *)
+          .returning(createInboundResultError(InvalidUserType(None)))
+
+        val res = controller
+          .view()
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:150]:
```suggestion
          .apply(buildFakeRequestWithSessionId("", ""))
```

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 6]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+            _: RequestWithUserDetails[AnyContent],
+            _: HeaderCarrier,
+            _: ExecutionContext
+          ))
+          .expects(List(SA_ASSIGNED_TO_OTHER_USER), *, *, *)
+          .returning(createInboundResultError(InvalidUserType(Some("/test"))))
+
+        val result = controller
+          .view()
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:116]:
```suggestion
          .apply(buildFakeRequestWithSessionId("", ""))
```

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 7]
```diff
@@ -35,26 +39,42 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+            _: HeaderCarrier,
+            _: ExecutionContext
+          ))
+          .expects(List(SA_ASSIGNED_TO_OTHER_USER), *, *, *)
+          .returning(createInboundResult(SA_ASSIGNED_TO_OTHER_USER))
+
         val result = controller
           .view()
-          .apply(request)
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:77]:
```suggestion
          .apply(buildFakeRequestWithSessionId("", "))
```

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 8]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
           .text() shouldBe "selfAssessmentInterrupt.paragraph1 " ++ "selfAssessmentInterrupt.paragraph2"
       }
     }
+
+    "the user does not have an account type of SA_ASSIGNED_TO_OTHER_USER" should {
+      "redirect to account check" in {
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:92]: controller reverse route

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 9]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+          .apply(buildFakePOSTRequestWithSessionId(Map.empty))
+
+        status(result) shouldBe SEE_OTHER
+        redirectLocation(result) shouldBe Some(
+          "/protect-tax-info/enrol-pt/other-user-id-has-sa/keep-access-to-sa-from-pta"
+        )
+      }
+    }
+
+    "the user does not have an account type of SA_ASSIGNED_TO_OTHER_USER" should {
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:193]: enum

File: test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala [hunk 10]
```diff
@@ -67,5 +87,173 @@ class SABlueInterruptControllerSpec extends TestFixture {
...
+          .view()
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
+
+        status(res) shouldBe OK
+        contentAsString(res) should include("enrolmentError.title")
+      }
+    }
+  }
+
+  "post" when {
```
Comment (ChrisHillCap) [inline test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/SABlueInterruptControllerSpec.scala:158]: being picky but we arent actually posting

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/forms/KeepAccessToSAThroughPTAForm.scala [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+import play.api.data.Form
+import play.api.data.Forms.{default, mapping, text}
+import play.api.data.validation.Constraints._
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.models.forms.{
+  EnrolCurrentUserId,
+  KeepAccessToSAThroughPTA
+}
+
+object KeepAccessToSAThroughPTAForm {
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/forms/KeepAccessToSAThroughPTAForm.scala:27]: Theres no form spec

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]

File: conf/application.conf [hunk 12]
```diff
@@ -104,7 +104,7 @@ microservice {
       host = "http://localhost:9232"
     }
 
-    bas-gateway {
+    bas-gateway-frontend {
```
Comment (ChrisHillCap) [inline conf/application.conf:107]: does app-config-qa, app-config-staging & app-config-production need to be amended or removed?

**Review by ChrisHillCap — DISMISSED** [review]

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/KeepAccessToSAController.scala [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+              case Right(call) =>
+                sessionCache
+                  .save[KeepAccessToSAThroughPTA](
+                    KEEP_ACCESS_TO_SA_THROUGH_PTA_FORM,
+                    keepAccessToSA
+                  )
+                  .map(_ => Redirect(call))
+              case Left(error) =>
+                Future.successful(
+                  handleErrors(error, "[KeepAccessToSAController][continue]")
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/KeepAccessToSAController.scala:114]: if handle errors logs then cool, but if not, dont we want to log here

Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/KeepAccessToSAController.scala:114]: Yes it does log, we pass the string in to add to the logs :-)

File: conf/application.conf (hunk 12 above)
Comment (Chrissystk) [reply conf/application.conf:107]: In the comments for this PR there are the other PRs that need to be merged when this one is. (app-config-base and the journey tests). The app-config base has the change.

**Comment by Chrissystk** [comment]
@ChrisHillCap I've addressed the comments

**Review by TSiyamachiraQA — APPROVED** [review]

**Review by TSiyamachiraQA — DISMISSED** [review]

**Review by TSiyamachiraQA — APPROVED** [review]

**Pull request merged**